scripts/run_mypy_checks.py

```diff
@@ -0,0 +1,810 @@
+    'scripts/regenerate_requirements_test.py',
+    'scripts/release_scripts/cut_release_or_hotfix_branch.py',
+    'scripts/release_scripts/cut_release_or_hotfix_branch_test.py',
+    'scripts/release_scripts/repo_specific_changes_fetcher.py',
+    'scripts/release_scripts/repo_specific_changes_fetcher_test.py',
+    'scripts/release_scripts/update_changelog_and_credits.py',
+    'scripts/release_scripts/update_changelog_and_credits_test.py',
+    'scripts/release_scripts/update_configs.py',
+    'scripts/release_scripts/update_configs_test.py',
+    'scripts/run_backend_tests.py',
+    'scripts/run_custom_eslint_tests.py',
+    'scripts/run_e2e_tests.py',
+    'scripts/run_e2e_tests_test.py',
+    'scripts/run_frontend_tests.py',
+    'scripts/run_lighthouse_tests.py',
+    'scripts/run_mypy_checks.py',
+    'scripts/run_mypy_checks_test.py',
+    'scripts/run_portserver.py',
+    'scripts/run_presubmit_checks.py',
+    'scripts/run_tests.py',
+    'scripts/script_import_test.py',
+    'scripts/scripts_test_utils.py',
+    'scripts/scripts_test_utils_test.py',
+    'scripts/servers.py',
+    'scripts/servers_test.py',
+    'scripts/setup.py',
+    'scripts/setup_gae.py',
+    'scripts/setup_gae_test.py',
+    'scripts/setup_test.py',
+    'scripts/start.py',
+    'scripts/third_party_size_check.py',
+    'scripts/typescript_checks.py',
+    'scripts/typescript_checks_test.py',
+]
+
+
+CONFIG_FILE_PATH = os.path.join('.', 'mypy.ini')
+# TODO(#13113): Change mypy command to mypy path after Python3 migration.
+MYPY_CMD = 'mypy'
+MYPY_REQUIREMENTS_PATH = os.path.join('.', 'mypy_requirements.txt')
+PYTHON3_CMD = 'python3'
+
+
+_PARSER = argparse.ArgumentParser(
+    description='Python type checking using mypy script.'
+)
+
+_PARSER.add_argument(
+    '--files',
+    help='Files to type-check',
+    action='store',
+    nargs='+'
+)
+
+
+def get_mypy_cmd(files):
+    """Return the appropriate command to be run.
+
+    Args:
+        files: list(list(str)). List having first element as list of string.
+
+    Returns:
+        list(str). List of command line arguments.
+    """
+    if files:
+        cmd = [MYPY_CMD, '--config-file', CONFIG_FILE_PATH] + files
+    else:
+        excluded_files_regex = (
+            '|'.join(NOT_FULLY_COVERED_FILES + EXCLUDED_DIRECTORIES))
+        cmd = [
+            MYPY_CMD, '--exclude', excluded_files_regex,
+            '--config-file', CONFIG_FILE_PATH, '.'
+        ]
+    return cmd
+
+
+def install_mypy_prerequisites():
+    """Install mypy and type stubs from mypy_requirements.txt.
+
+    Returns:
+        int. The return code from installing prerequisites.
+    """
+    cmd = [PYTHON3_CMD, '-m', 'pip', 'install', '-r', MYPY_REQUIREMENTS_PATH]
+    process = subprocess.call(
+        cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+    return process
+
+
+def main(args=None):
+    """Runs the MyPy type checks."""
+    parsed_args = _PARSER.parse_args(args=args)
+
+    install_third_party_libs.main()
```

Comment:
@DubeySandeep Do you think this should be outside main? For others like `run_backend_tests` I see they are written at the [top](https://github.com/oppia/oppia/blob/140c2919b6a8ba5e28c701558bf0fa2efc76d485/scripts/run_backend_tests.py#L69).

Reply:
@sajalasati I had a look at pre_commit_linter and we install the libraries inside main. Thanks!

Reply:
It is required on top, if the script needs to import something which is expected to install via install_third_party_libs. [Make sure that we don't need to install any third_party to import python_utils on top! Maybe test the run_mypy script after clearing oppia_tools and third-party dir. (OR move thises folders to a temporary dir and then run the script to check whether it starts downloading third_party_libs)]

Reply:
When I ran the workflow without install_third_party_libs and an error occured saying 'yaml' was not found. This occured due to import python_utils. What should be done here? I am trying to mimic the way lint.yml is written.

Reply:
> When I ran the workflow without install_third_party_libs and an error occured saying 'yaml' was not found. This occured due to import python_utils. What should be done here? I am trying to mimic the way lint.yml is written.

@DubeySandeep Then it seems we do need to have third party libs installed for python_utils to be imported without errors. So I would suggest @hardikkat24 to shift `install_third_party_libs.main()` to the top (like the example I gave here) and also keep `run: python -m scripts.install_third_party_libs ` in the yml file for now. We can perhaps create new issue to solve this and get this PR merged ASAP - @DubeySandeep will you do this?

Reply:
@sajalasati I have moved install_third_party_libs to the top. PTAL.